parser.py

```diff
@@ -45,11 +45,22 @@ def format_wires(wire_id_list):
 
 class ArithmeticOp:
 
+    def __new__(cls, func, *exps):
+        if len(exps) == 1:
+            return super().__new__(UnaryOp)
+
+        if len(exps) == 2:
+            return super().__new__(BinaryOp)
```

Comment:
Moves the logic on whether to instantiate a UnaryOp or a BinaryOp into the Arithmetic class itself, to avoid having to replicate it everywhere